lib/mayaUsd/fileio/jobs/readJob.cpp

```diff
@@ -602,85 +844,83 @@ void UsdMaya_ReadJob::_ImportPrototype(
 
-        _PrimReaderMap     primReaderMap;
-        const UsdPrimRange range = buildInstances
+        const UsdPrimRange range = mArgs.importInstances
             ? UsdPrimRange::PreAndPostVisit(rootPrim)
             : UsdPrimRange::PreAndPostVisit(
                 rootPrim, UsdTraverseInstanceProxies(UsdPrimAllPrimsPredicate));
-
-        const int                     loopSize = std::distance(range.begin(), range.end());
-        MayaUsd::ProgressBarLoopScope instanceLoop(loopSize);
-        for (auto primIt = range.begin(); primIt != range.end(); ++primIt) {
-            const UsdPrim&           prim = *primIt;
-            UsdMayaPrimReaderContext readCtx(&mNewNodeRegistry);
-            readCtx.SetTimeSampleMultiplier(mTimeSampleMultiplier);
-
-            if (buildInstances && prim.IsInstance()) {
-                _DoImportInstanceIt(primIt, usdRootPrim, readCtx, primReaderMap);
-            } else {
-                _DoImportPrimIt(primIt, usdRootPrim, readCtx, primReaderMap);
-            }
-            instanceLoop.loopAdvance();
-        }
+        _ImportPrimRange(range, usdRootPrim);
     }
 
-    if (buildInstances) {
-        progressBar.addSteps(1);
+    return _CleanupPrototypes(usdRootPrim);
+}
+
+void UsdMaya_ReadJob::_ImportPrimRange(const UsdPrimRange& range, const UsdPrim& usdRootPrim)
+{
+    const int                     loopSize = std::distance(range.begin(), range.end());
+    MayaUsd::ProgressBarLoopScope instanceLoop(loopSize);
+
+    _PrimReaderMap primReaderMap;
 
-        MDGModifier              deletePrototypeMod;
+    for (auto primIt = range.begin(); primIt != range.end(); ++primIt) {
+        const UsdPrim&           prim = *primIt;
         UsdMayaPrimReaderContext readCtx(&mNewNodeRegistry);
         readCtx.SetTimeSampleMultiplier(mTimeSampleMultiplier);
 
-        auto                          prototypes = usdRootPrim.GetStage()->GetPrototypes();
-        const int                     loopSize = prototypes.size();
-        MayaUsd::ProgressBarLoopScope prototypesLoop(loopSize);
-        for (const auto& prototype : prototypes) {
-
-            const SdfPath prototypePath = prototype.GetPath();
-            MObject       prototypeObject = readCtx.GetMayaNode(prototypePath, false);
-            if (prototypeObject != MObject::kNullObj) {
-                MStatus    status;
-                MFnDagNode prototypeNode(prototypeObject, &status);
-                if (status) {
-                    while (prototypeNode.childCount()) {
-                        prototypeNode.removeChildAt(prototypeNode.childCount() - 1);
-                    }
+        if (mArgs.importInstances && prim.IsInstance()) {
+            _DoImportInstanceIt(primIt, usdRootPrim, readCtx, primReaderMap);
+        } else {
+            _DoImportPrimIt(primIt, usdRootPrim, readCtx, primReaderMap);
+        }
+        instanceLoop.loopAdvance();
+    }
+}
+
+bool UsdMaya_ReadJob::_CleanupPrototypes(const UsdPrim& usdRootPrim)
```

Comment:
Ignore whitespace difference will simplify the diff here. Most code is the same, just with one level less of indentation.